spec-tests/src/test/java/org/candlepin/spec/ExportSpecTest.java

```diff
@@ -0,0 +1,621 @@
+        JsonNode exportIdNode = root.get("exportId");
+        if (exportIdNode == null) {
+            return null;
+        }
+
+        return apiClient.consumers().downloadExistingExport(consumerUuid, exportIdNode.textValue());
+    }
+
+    private ConsumerDTO createReadOnlyConsumer(ApiClient client, OwnerDTO owner) throws ApiException {
+        UserDTO readOnlyUser = UserUtil.createReadOnlyUser(client, owner);
+        ConsumerDTO readOnlyConsumer = Consumers.random(owner, ConsumerTypes.Candlepin);
+
+        return client.consumers().createConsumer(readOnlyConsumer, readOnlyUser.getUsername(),
+            owner.getKey(), null, true);
+    }
+
+    private ProductDTO createVDCProduct(ApiClient client, String ownerKey) throws ApiException {
+        OwnerProductApi ownerProductsApi = client.ownerProducts();
+        ProductDTO productDc = Products.withAttributes(ProductAttributes.Arch.withValue("x86_64"),
+            ProductAttributes.Stacking.withValue("stack-dc"));
+        productDc = ownerProductsApi.createProductByOwner(ownerKey, productDc);
+
+        ProductDTO productVdc = Products.withAttributes(ProductAttributes.Arch.withValue("x86_64"),
+            ProductAttributes.VirtualLimit.withValue("unlimited"),
+            ProductAttributes.Stacking.withValue("stack-vdc"));
+        productVdc.setDerivedProduct(productDc);
+        productVdc = ownerProductsApi.createProductByOwner(ownerKey, productVdc);
+
+        return productVdc;
+    }
+
+    private List<JsonNode> bindPoolsToConsumer(ConsumerClient consumerApi, String consumerUuid,
+        Collection<String> poolIds) throws ApiException, JsonProcessingException {
+        List<JsonNode> poolNodes = new ArrayList<>();
+        for (String poolId : poolIds) {
+            poolNodes.add(consumerApi.bindPool(consumerUuid, poolId, 1));
+        }
+
+        return poolNodes;
+    }
+
+    private Map<String, PoolDTO> createPoolsForProducts(OwnerClient ownerApi, String ownerKey,
+        Collection<ProductDTO> products, Collection<BrandingDTO> brandings) throws ApiException {
+        Map<String, PoolDTO> poolIdToPool = new HashMap<>();
+        for (ProductDTO product : products) {
+            PoolDTO pool = createPool(ownerApi, ownerKey, product, 2L, brandings);
+            poolIdToPool.put(pool.getId(), pool);
+        }
+
+        return poolIdToPool;
+    }
+
+    private PoolDTO createPool(OwnerClient ownerApi, String ownerKey, ProductDTO product, long quantity,
+        Collection<BrandingDTO> brandings) throws ApiException {
+        PoolDTO pool = Pools.random(product)
+            .quantity(quantity)
+            .providedProducts(new HashSet<>())
+            .accountNumber("12345")
+            .orderNumber("6789")
+            .endDate(OffsetDateTime.now().plusYears(5))
+            .branding(new HashSet<>(brandings));
+
+        return ownerApi.createPool(ownerKey, pool);
+    }
+
+    private void waitForJob(ApiClient client, String jobId) throws IOException {
```

Comment:
I added this method because the `JobsUtil.waitForJob()` method throws errors for shouldPurgeImportRecords test when trying to retrieve the job status and de-serializing the job status object. So, to avoid this, I'm using the Request api and just reading the state of the job off the json.

Reply:
What exception are you seeing?

Reply:
This is that issue where the import job is returning an object in the resultData field for the AsyncJobStatusDTO instead of a string, and so the ApiClient can't deserialize the json into the AsyncJobStatusDTO.

```
ExportSpecTest > Standard Export Tests > should purge import records FAILED
    java.lang.IllegalArgumentException: Expected the field `resultData` to be a primitive type in the JSON string but got `{"created":1662061459887,"updated":1662061459887,"id":"ff80818182fa8d340182fa9539af3072","status":"SUCCESS","statusMessage":"test_owner-97717E27 file imported forcibly.","fileName":"8991662c-124e-41c8-a629-bbcb6e3c87f9-export-11672463340225012102.zip","generatedBy":"admin","generatedDate":1662061459092,"upstreamConsumer":{"created":1662061459887,"updated":1662061459887,"id":"ff80818182fa8d340182fa9539af3073","uuid":"8991662c-124e-41c8-a629-bbcb6e3c87f9","name":"test_consumer-0413CED7","type":{"id":"1003","label":"candlepin","manifest":true},"ownerId":"ff80818182fa8d340182fa9538423045","contentAccessMode":"org_environment","webUrl":"cdn-name80224C92","apiUrl":"https://cdn.test.com"}}`
        at org.candlepin.dto.api.v1.AsyncJobStatusDTO.validateJsonObject(AsyncJobStatusDTO.java:670)
        at org.candlepin.dto.api.v1.AsyncJobStatusDTO$CustomTypeAdapterFactory$1.read(AsyncJobStatusDTO.java:695)
        at org.candlepin.dto.api.v1.AsyncJobStatusDTO$CustomTypeAdapterFactory$1.read(AsyncJobStatusDTO.java:685)
        at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199)
        at com.google.gson.Gson.fromJson(Gson.java:1058)
        at com.google.gson.Gson.fromJson(Gson.java:1016)
        at com.google.gson.Gson.fromJson(Gson.java:959)
        at org.candlepin.JSON.deserialize(JSON.java:246)
        at org.candlepin.ApiClient.deserialize(ApiClient.java:847)
        at org.candlepin.ApiClient.handleResponse(ApiClient.java:1055)
        at org.candlepin.ApiClient.execute(ApiClient.java:979)
        at org.candlepin.resource.JobsApi.getJobStatusWithHttpInfo(JobsApi.java:541)
        at org.candlepin.resource.JobsApi.getJobStatus(JobsApi.java:519)
        at org.candlepin.spec.bootstrap.client.api.JobsClient.waitForJob(JobsClient.java:66)
        at org.candlepin.spec.bootstrap.client.api.JobsClient.waitForJob(JobsClient.java:107)
        at org.candlepin.spec.ExportSpecTest$StandardExporter.shouldPurgeImportRecords(ExportSpecTest.java:297)

```

Reply:
This looks like something is wrong in the spec file or the client generation. That field should be a generic string at the API level. Before we work around it, we should check if there's something we can fix there to make it so we don't need to give it special attention.

Reply:
I was able to fix and validate the issue by changing the resultData field type to an object in the candlepin-api-spec.yaml file. This would also be nice because we can avoid adding all the multipart/form-data functionality to the Request object. 

https://github.com/candlepin/candlepin/blob/master/api/candlepin-api-spec.yaml#L8239

A side note, I also ran into this issue with the async export job as the result data is an ExportResult object instead of String. We can add a ExportResultDTO object to candlepin-api-spec.yaml and use the ObjectMapper to deserialize the resultData field to the ExportResultDTO.

Reply:
@Ceiu I did some validating in Candlepin 4.0 to verify that the changes to the resultData field are passive. I triggered an async export job using the `/consumers/{consumer_uuid}/export/async` endpoint and then got the job status using the `/jobs/{job_id}` endpoint and validated that the resultData field has an object and not a string.

CP 4.0job status result for async export:
```
{
  "created" : "2022-09-13T20:58:30+0000",
  "updated" : "2022-09-13T20:58:30+0000",
  "id" : "ff80818183389410018338a5720e0034",
  "name" : "Export Manifest",
  "group" : null,
  "origin" : "candlepin-el8.example.com",
  "executor" : "candlepin-el8.example.com",
  "principal" : "admin",
  "state" : "FINISHED",
  "previousState" : "RUNNING",
  "startTime" : "2022-09-13T20:58:30+0000",
  "endTime" : "2022-09-13T20:58:30+0000",
  "attempts" : 1,
  "maxAttempts" : 1,
  "statusPath" : "/jobs/ff80818183389410018338a5720e0034",
  "resultData" : {"exportedConsumer":"35398b97-11f2-428c-a3e5-fa286bdaddc8","exportId":"ff80818183389410018338a572dd0035","href":"/consumers/35398b97-11f2-428c-a3e5-fa286bdaddc8/export/ff80818183389410018338a572dd0035"},
  "key" : "ExportJob"
}
```